gadget-container/gadgettracermanager/auth.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+	"k8s.io/client-go/kubernetes"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+
+	certhelpers "github.com/inspektor-gadget/inspektor-gadget/internal/cert-helpers"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/certs"
+)
+
+func streamInterceptor() func(srv interface{}, serverStream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	return func(srv interface{}, serverStream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+		p, ok := peer.FromContext(serverStream.Context())
+		if !ok {
+			return status.Errorf(codes.Unauthenticated, "invalid certificate")
+		}
+		tlsInfo := p.AuthInfo.(credentials.TLSInfo)
+		subject := tlsInfo.State.VerifiedChains[0][0].Subject
+		log.Infof("stream request from %s", subject)
+		return handler(srv, serverStream)
+	}
+}
+
+func unaryInterceptor() func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
```

Comment:
I'm wondering if this would make sense to declare function directly and avoid returning a lambda? 

```
func unaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
	p, ok := peer.FromContext(ctx)
	if !ok {
		return nil, status.Errorf(codes.Unauthenticated, "invalid certificate")
	}
	tlsInfo := p.AuthInfo.(credentials.TLSInfo)
	subject := tlsInfo.State.VerifiedChains[0][0].Subject
	log.Infof("request from %s", subject)
	return handler(ctx, req)
}
```

And use `unaryInterceptor` instead of `unaryInterceptor()` in main.go.

Reply:
Yeah, I had those for token authentication earlier where I would use the outer function to have that as its parameter - doesn't make sense this way for these functions, but they're only here for debugging right now (but we could use it to prepend the logs with the CN of the incoming certificate for example).